src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -260,6 +217,16 @@ public TopicPartitionChannel(
             && this.conn != null
             && this.conn.hasSchemaEvolutionPermission(
                 tableName, sfConnectorConfig.get(SNOWFLAKE_ROLE));
+
+    // Open channel and reset the offset in kafka
+    this.channel = Preconditions.checkNotNull(openChannelForTable());
+    final long lastCommittedOffsetToken = fetchOffsetTokenWithRetry();
+    this.offsetPersistedInSnowflake.set(lastCommittedOffsetToken);
+    this.processedOffset.set(lastCommittedOffsetToken);
+    if (lastCommittedOffsetToken != NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
+      // Reset the offset in kafka, only if we have a valid offset token at server side
+      this.sinkTaskContext.offset(this.topicPartition, lastCommittedOffsetToken);
```

Comment:
I can see a potential exception happening here from kafka side which somwhat says `offset not present` when we try to reset an offset which is not present. Lets say for example a topic is deleted. 

This will force them to delete the table because there is no way to force reset an offset on snowflake side.

Reply:
Also, 
your changes are doing exactly what the docs says
```
When the topic partition is recovered the task would reload offsets from HDFS and use this method to reset the consumer to the offset
```
👍

Reply:
> I can see a potential exception happening here from kafka side which somwhat says offset not present when we try to reset an offset which is not present. Lets say for example a topic is deleted.

Good point, any suggestion on what we should do here? Without my change, we will just accept whatever kafka sends us?